%description:
test cQueue::Iterator with empty queue
test iter++, iter-- and *iter behaves good after reaching end()

%activity:

#define CHECK(cond)  if (!(cond)) {throw cRuntimeError("BUG at line %d, failed condition %s", __LINE__, #cond);}

cQueue q("queue");

cQueue::Iterator iter(q);
CHECK(iter.end());
CHECK(*iter==nullptr);

iter++;
CHECK(iter.end());
CHECK(*iter==nullptr);

iter++;
CHECK(iter.end());
CHECK(*iter==nullptr);

iter--;
CHECK(iter.end());
CHECK(*iter==nullptr);

iter--;
CHECK(iter.end());
CHECK(*iter==nullptr);

// now with reverse==true
cQueue::Iterator iter2(q,true);
CHECK(iter2.end());
CHECK(*iter2==nullptr);

iter2++;
CHECK(iter2.end());
CHECK(*iter2==nullptr);

iter2--;
CHECK(iter2.end());
CHECK(*iter2==nullptr);

EV << "OK!\n";

%contains: stdout
OK!

%not-contains: stdout
BUG

